liblll/CompilerState.cpp

```diff
@@ -54,8 +54,8 @@ void CompilerState::populateStandard()
 	"(def 'msg (gaslimit to value data) { [0]:data (msg gaslimit to value 0 32) })"
 	"(def 'msg (to value data) { [0]:data (msg allgas to value 0 32) })"
 	"(def 'msg (to data) { [0]:data (msg allgas to 0 0 32) })"
-	"(def 'create (value code) { [0]:(msize) (create value @0 (lll code @0)) })"
-	"(def 'create (code) { [0]:(msize) (create 0 @0 (lll code @0)) })"
+	"(def 'create (value code) { [0]:{[0]:0xff (msize)} (create value @0 (lll code @0)) })"
+	"(def 'create (code) { [0]:{[0]:0xff (msize)} (create 0 @0 (lll code @0)) })"
```

Comment:
Consider the original code when `msize` is zero at first:

```
  mstore(0x0, msize) <-- inserts 0 in M[0] (memory location 0)
  dataSize(sub_0)
  dup1
  dataOffset(sub_0)
  mload(0x0)  <-- This is 0
  codecopy  <-- Overwrites M[0] with the first word of sub_0!!! This is a bug.
  mload(0x0) <-- Ought to be zero, but is now probably not.
  0x0
  create <-- this will produce garbage (memory access violation, probably)
stop
```

In the second case, again with msize being zero at first:

```
 mstore(0x0, 0xff)  <-- This is a hack to ensure that msize is >=32. (When msize is already >=32 this has no effect.)
  mstore(0x0, msize) <-- now msize=32 so M[0] is set to 32
  dataSize(sub_0)
  dup1
  dataOffset(sub_0)
  mload(0x0)
  codecopy <-- copies to memory starting with M[32]. This is fine.
  mload(0x0) <-- M[0] is still 32. This is good!
  0x0
  create <-- will now do the right thing.
stop
```

Admittedly, doing the initial `[0]:0xff` is a bit hacky and obscure. Really we should set M[0] to be `(min (msize) 32)`, but we have no `min` macro, and it would take a load more opcodes and a jump. So this is the best quick fix I could come up with that just fixed this edge case but had no effect in the general case.

Another option is to change the macro to 

```
(def 'create (code) { [0]:(+ 32 (msize)) (create 0 @0 (lll code @0)) })
```

But unlike my fix that affects all cases, not just when msize<32 to start with.

I hope that is clearer. Let me know if you need more info.